compiler/rustc_session/src/config.rs

```diff
@@ -1339,7 +1339,8 @@ pub fn rustc_short_optgroups() -> Vec<RustcOptGroup> {
         opt::opt_s(
             "",
             "edition",
-            "Specify which edition of the compiler to use when compiling code.",
+            Box::leak(format!("Specify which edition of the compiler to use when compiling code.\
```

Comment:
Could you please avoid this `Box::leak` call? It is possible to invoke rustc multiple times in the same process. Each time would leak memory. You should be able to inline the `opt_s` function to `pub fn opt_s(a: S, b: S, c: &str, d: S) -> R {` to allow strings with arbitrary lifetimes and then pass `&format(...)`.

Reply:
Yep - I was actually pretty wary of this change but that's what I was recommended to do. I'll rewrite the type signature and see if it still works soon so we can avoid this leak.

Reply:
Unfortunately I cannot refactor this out as the reference needs to outlive `'static`

```
error[E0521]: borrowed data escapes outside of function
    --> compiler/rustc_session/src/config.rs:1290:9
     |
1289 |     pub fn opt_s(a: S, b: S, c: &str, d: S) -> R {
     |                  -           -  - let's call the lifetime of this reference `'1`
     |                  |           |
     |                  |           `c` is a reference that is only valid in the function body
     |                  `a` declared here, outside of the function body
1290 |         stable(longer(a, b), move |opts| opts.optopt(a, b, c, d))
     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |         |
     |         `c` escapes the function body here
     |         argument requires that `'1` must outlive `'static`
```

There's got to be something I'm missing - the enum variant for the formatted string are all static and embedded, and there's got to be a better option than either allocating, or just manually writing in the strings and commenting above the original variables, "this must be updated in xyz string". I don't know how but I think there might be a way to construct this statically somehow - it's just concatenating all-static strings.

Reply:
I see. Another option would be to use `std::sync::LazyLock` to compute the description. If you put it in a static it should allow you to get a single `&'static str` without leaking anything.